### ISSUE TITLE
DX12: Saint Seiya Brave Soldiers - fragment program compile failure

### ISSUE THREAD

**Issue by raven02** [RPCS3/rpcs3#2050]
DX12 (Stuck at "Now Loading" scene)

```
E {rsx::thread} RSX: FS build failed:D:\EMU\PS3\RPCS3.V2\FragmentProgram.hlsl(197,12-55): warning X3570: gradient instruction used in a loop with varying iteration, attempting to unroll the loop
D:\EMU\PS3\RPCS3.V2\FragmentProgram.hlsl(188,3-36): error X3511: unable to unroll loop, loop does not appear to terminate in a timely manner (248 iterations) or unrolled loop is too large, use the [unroll(n)] attribute to force an exact higher number

F {rsx::thread} class std::runtime_error thrown: fragment program compilation failure
(in file c:\rpcs3\rpcs3\emu\rsx\d3d12\D3D12PipelineState.h:162)

```

![untitled](https://cloud.githubusercontent.com/assets/3000282/17876376/3d8b1998-690c-11e6-88ab-d2e9ff1759d5.jpg)

**Comment by ikki84** [comment]
@raven02  OpenGL and Vulkan goes further now (tested with fcb85379)
https://ci.appveyor.com/api/buildjobs/bvjdcpmxcq0nb0j2/artifacts/rpcs3-ReleaseLLVM-fcb85379.zip

![ssbs2](https://cloud.githubusercontent.com/assets/16824659/18816744/cfc2e286-8325-11e6-8549-dfdd74ffd713.jpg)
![ssbs3](https://cloud.githubusercontent.com/assets/16824659/18816745/cff82e46-8325-11e6-92a5-cb72def6bde0.jpg)

-The black shadow at the side of the clothes appears to be  an attempt to render the model without the clothes...
-DX12 still stucks at the loading screen

**Comment by ikki84** [comment]
It´s affects (Saint Seiya - Soldier´s Soul [BLES01421]) the same way. 

*off-topic* - the game (Saint Seiya: Sanctuary Battle [BLES-01421]) is running in the last builds  (i don´t know since when it start running), and the characters clothes are all kind of white (in OpenGL). Maybe its also related to the issue #2178.
Vulkan and DX12 is a totally mess on the screen.

@raven02 @kd-11

**Comment by raven02** [comment]
Just wonder which LLE is required to in-game .

**Comment by ikki84** [comment]
I will list here all that are selected, but i think that not all are necessary:

*Also to solve the fonts problem, I used the fonts files in "/dev_flash/data/font"* (!)

-libadec
-libdmux
-libdmuxpamf
-libfiber
-libfonts          <-  (!)
-libfontsFT      <-  (!)
-libfreetype     <-  (!)
-libfreetypeTT <-  (!)
-libfs
-lib10n
-libpamf
-libresc
-librtc
-libsail
-libsail_avi
-libsail_rec
-libsmvd2
-libsmvd4
-libspurs_jq
-libsre
-libssl
-libsync2
-libvdec

@raven02

**Comment by raven02** [comment]
Thanks @ikki84

**Comment by ikki84** [comment]
@raven02 Thank you and all the developers of rpcs3. 
The emulator is making progress :)

**Comment by ikki84** [comment]
I was testing this game again, the last build it was working is:

RPCS3 v0.0.1-3-8454949 Pre-Alpha:
https://ci.appveyor.com/api/buildjobs/hb0slyqiyw809da8/artifacts/rpcs3-ReleaseLLVM-8454949e.zip

Since then all the master builds stopped working, using the lle modules that I listed above.

In a build that is not merged yet it became to work again with some regressions: 

RPCS3 v0.0.1-3-123f51d Pre-Alpha
https://ci.appveyor.com/api/buildjobs/7qol4mha9p7iur94/artifacts/rpcs3-ReleaseLLVM-123f51d5.zip

*The title screen miss the background  (looks like the front part its hidding it) and the saint´s clothes are flickering and changing position all the time.


RPCS3 v0.0.1-3-8454949 Pre-Alpha:

![title screen rpcs3 v0 0 1-3-8454949 pre-alpha](https://cloud.githubusercontent.com/assets/16824659/19629625/b449108c-9957-11e6-86e0-8edcfeca00ef.jpg)
![character select rpcs3 v0 0 1-3-8454949 pre-alpha](https://cloud.githubusercontent.com/assets/16824659/19629638/cd45657c-9957-11e6-872d-1be6842a432a.jpg)


RPCS3 v0.0.1-3-123f51d Pre-Alpha:

![title screen - rpcs3 v0 0 1-3-123f51d pre-alpha](https://cloud.githubusercontent.com/assets/16824659/19629624/b43f57ae-9957-11e6-8ef9-6b482d6f4883.jpg)
![character select - 1 rpcs3 v0 0 1-3-123f51d pre-alpha](https://cloud.githubusercontent.com/assets/16824659/19629637/cd3d4afe-9957-11e6-8913-550189b9eac0.jpg)
![character select -2 rpcs3 v0 0 1-3-123f51d pre-alpha](https://cloud.githubusercontent.com/assets/16824659/19629636/cd3cd54c-9957-11e6-9ca0-184908e55eda.jpg)


Logs:
[log RPCS3 v0.0.1-3-8454949 Pre-Alpha.txt](https://github.com/RPCS3/rpcs3/files/546651/log.RPCS3.v0.0.1-3-8454949.Pre-Alpha.txt)
[log RPCS3 v0.0.1-3-123f51d Pre-Alpha.txt](https://github.com/RPCS3/rpcs3/files/546652/log.RPCS3.v0.0.1-3-123f51d.Pre-Alpha.txt)

**Comment by kd-11** [comment]
Some bugs here may be fixed now since https://github.com/RPCS3/rpcs3/pull/2237
Missing geometry may be due to previously broken handling of parallel branch instructions. These can be spotted by searching for "jump_position" in the vertex shaders or the log. This is one such case.

**Comment by SakataGintokiYT** [comment]
@kd-11 
much better :)
<img src="http://www.fotoszok.pl/upload/510ca43e.jpg" alt="Fotoszok.pl" />
`E {rsx::thread} RSX: Unimplemented sca_opcode BRB`

**Comment by SakataGintokiYT** [comment]
@kd-11 
[9e84804e]
Saint Seiya Brave Soldiers:

**[Log:](https://gist.github.com/SakataGintokiYT/6afbfbfc2d2bbac8a5e07b63f0f53c6f)**
**[shaderLog:](https://mega.nz/#!FEJBzbCK!baakWFcrUhO5fDhxUu-LCSHTBdm9APoz_41QtSJfvu4)**
**[OpenGL-RenderDoc-Menu:](https://mega.nz/#!oYgXWCoD!d6lJgIxRjH5zNrasTDAFudyL1ChRRdvuXExoevc0110)**
**[OpenGL-RenderDoc-In game:](https://mega.nz/#!kEQFCLgT!gra38HAM6yBDFppErioELjDObo8UUTVO5pIqoLvfh0U)**

**Comment by kd-11** [comment]
Wait; so the armor was rendered correcly before (i.e not flashing and misplaced)?
If so, I might just have to reorganize the code generation and this might be fixed.

Anyway, the generated shader now has alot of dead code; Not sure if its supposed to be like this.

**Comment by kd-11** [comment]
I suspect the unimplemented branch instruction is what keeps some code from being executed at all, causing the flicker.

**Comment by kd-11** [comment]
https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.378
Post the emulator log with this build; It should log the call to the undefined op and give some more information.

**Comment by SakataGintokiYT** [comment]
@kd-11 
[6f6d0dca]

**[Log:](https://gist.github.com/SakataGintokiYT/20cf0a82a245cc1b39d28e8ad16ae32a)**
**[shaderLog:](https://mega.nz/#!NJIBzbTR!rj-Iu7Xl1uHpztuFJrQXX4cQ8dbSxZ5roWESAHHqg7A)**
**[OpenGL-RenderDoc-Menu:](https://mega.nz/#!1QRiBbRL!GC7d4pVM7obZFrs5W_O0c7KM_Ky21H0KVPqOlQgur-w)**
**[OpenGL-RenderDoc-character select screen:](https://mega.nz/#!EZBhCbzB!xM1uRJkUizAYKRxbOkSaGq-sEZZZJLqDnOadTKsP-ag)**
**[OpenGL-RenderDoc-In game:](https://mega.nz/#!EdZABITQ!8ryKHKHe-FUHHrJBW-gn1WiUqKtjZ1-K_ZYO97DGcE0)**

**Comment by ikki84** [comment]
@kd-11  Yes, the armor was rendered correcly before.

**Comment by SakataGintokiYT** [comment]
@kd-11 
[6f6d0dca]

**[Log:](https://gist.github.com/SakataGintokiYT/20cf0a82a245cc1b39d28e8ad16ae32a)**
**[shaderLog:](https://mega.nz/#!NJIBzbTR!rj-Iu7Xl1uHpztuFJrQXX4cQ8dbSxZ5roWESAHHqg7A)**
**[OpenGL-RenderDoc-Menu:](https://mega.nz/#!1QRiBbRL!GC7d4pVM7obZFrs5W_O0c7KM_Ky21H0KVPqOlQgur-w)**
**[OpenGL-RenderDoc-character select screen:](https://mega.nz/#!EZBhCbzB!xM1uRJkUizAYKRxbOkSaGq-sEZZZJLqDnOadTKsP-ag)**
**[OpenGL-RenderDoc-In game:](https://mega.nz/#!EdZABITQ!8ryKHKHe-FUHHrJBW-gn1WiUqKtjZ1-K_ZYO97DGcE0)**


```
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C088, d3=0x20401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C088, d3=0x1401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C089, d3=0x61401FFC
E {rsx::thread} RSX: Unknown/illegal instruction: 0x3d (forced unit 2)
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C082, d3=0xA0401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C092, d3=0x40401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x9C6C, d1=0x8840000D, d2=0x8086C093, d3=0xA141FFFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C084, d3=0x20401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C085, d3=0x81401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C084, d3=0x20401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C085, d3=0x81401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C084, d3=0x20401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C085, d3=0x81401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C084, d3=0x20401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C085, d3=0x81401FFC

E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C082, d3=0x401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C082, d3=0x401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unknown/illegal instruction: 0x3d (forced unit 2)
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Unimplemented sca_opcode BRB, d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C086, d3=0xE1401FFC

```

**Comment by ikki84** [comment]
Surely the problem is after pull request #2153 and before #2237.

@SakataGintokiYT  @kd-11  Could you test it?
https://ci.appveyor.com/api/buildjobs/f6pe0p1t4wejeo8c/artifacts/rpcs3-ReleaseLLVM-025e3f04.zip

Merry Christmas.

**Comment by SakataGintokiYT** [comment]
@ikki84 
same

**Comment by kd-11** [comment]
As far as I know, BRB is not properly implemented yet. I'll get back to rpcs3 after the holidays. Cheers.

**Comment by ikki84** [comment]
@SakataGintokiYT Strange, I am sure it fix the armor and the characters.
Could try it again?
Just try it using OpenGL. / Vulkan and DX12 will not work.

https://ci.appveyor.com/api/buildjobs/f6pe0p1t4wejeo8c/artifacts/rpcs3-ReleaseLLVM-025e3f04.zip

**Comment by SakataGintokiYT** [comment]
@ikki84 
Still the same issue maybe it's drivers question ?  or gpu model...

**Comment by ikki84** [comment]
Radeon R7 200.

**Comment by kd-11** [comment]
@SakataGintokiYT Retry with this build: https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.379
Attempts to actually implement the opcodes BRB and CLB

**Comment by MarioSonic2987** [comment]
@kd-11 
GL:

```
·F {rsx::thread} class gl::glsl::link_exception thrown: linkage failed: 'Vertex info
-----------
0(115) : error C0000: syntax error, unexpected $undefined at token "<undefined>"
0(283) : error C1118: "break" not in loop
0(286) : error C0000: syntax error, unexpected '}' at token "}"
0(301) : error C0000: syntax error, unexpected '(', expecting "::" at token "("
```

Vulkan:
```
·E {rsx::thread} RSX: Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:116: '$' : unexpected token 
ERROR: 1 compilation errors.  No code generated.


·E {rsx::thread} RSX: 
·F {rsx::thread} class std::runtime_error thrown: Failed to compile vertex shader
(in file C:\rpcs3\rpcs3\Emu\RSX\VK\VKVertexProgram.cpp:391)
```
[Shaderlog Vulkan](https://drive.google.com/open?id=0B9LymxJkpT48YVUyTnlBUjFhRm8)
[Shaderlog GL](https://drive.google.com/open?id=0B9LymxJkpT48VnpiQlRhM3QxT3M)

**Comment by kd-11** [comment]
https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.381
I dont expect much progress, just the GL shaderlog will do. This is mainly for experimenting with the code generator and figuring out how jump addresses are supposed to be determined.

**Comment by MarioSonic2987** [comment]
```
F {PPU[0x70000000] Thread (main_thread) [0x0102c0a4]} struct vm::access_violation thrown: Access violation reading address 0x0
```
[Shaderlog GL](https://drive.google.com/open?id=0B9LymxJkpT48akNLUGdGU0ZVYTA)
[Log GL](https://drive.google.com/open?id=0B9LymxJkpT48Y2lydnRtY1RXb2M)

**Comment by kd-11** [comment]
https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.383
Should dump the last raw compiled vertex program as vertexprogram.raw (opengl only) so that I can work on the shader code without having the game. Just attach the opengl shaderlog for this one.

**Comment by MarioSonic2987** [comment]
[Shaderlog3 GL](https://drive.google.com/open?id=0B9LymxJkpT48dkN3MXVrMGQ5YmM)

**Comment by SakataGintokiYT** [comment]
[17ce61d5]

@kd-11 

**GL**
```
E {rsx::thread} RSX: Unimplemented sca_opcode BRB d0=0x1F9C6C, d1=0x8800000D, d2=0x8106C081, d3=0x41401FFC
E {rsx::thread} RSX: Failed to compile vertex shader: 0(94) : error C0000: syntax error, unexpected $undefined at token "<undefined>"
0(138) : error C1118: "break" not in loop
0(141) : error C0000: syntax error, unexpected '}' at token "}"
0(156) : error C0000: syntax error, unexpected '(', expecting "::" at token "("

F {rsx::thread} class gl::glsl::link_exception thrown: linkage failed: 'Vertex info
-----------
0(94) : error C0000: syntax error, unexpected $undefined at token "<undefined>"
0(138) : error C1118: "break" not in loop
0(141) : error C0000: syntax error, unexpected '}' at token "}"
0(156) : error C0000: syntax error, unexpected '(', expecting "::" at token "("
(0) : error C2003: incompatible options for link
'
```
**[ShaderLog:](https://mega.nz/#!YAhViBhD!yzm5cjU50GQXMA1NhViANo18HFZ_ShM3_8AijqJKLmc)**

**Comment by kd-11** [comment]
I know about the typo, its fixed here: https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.384

**Comment by kd-11** [comment]
https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.385 generates better code, although I've yet to see a case where the encapsulated code should be taken.

**Comment by SakataGintokiYT** [comment]
[21e3f316]
@kd-11 

**[Shaderlog:](https://mega.nz/#!hYxSjZhI!d6Iy11Qs1vaW4byLDtykZCCNoPA02hI6Je9Sb4rkGBc)**
**[Console dump:](https://mega.nz/#!lZhlkLpY!12nHLfTA_W87vR9eliUTibKKmiJ3IROztE89j_tfQrw)**

**Comment by kd-11** [comment]
I now understand how the BRB instruction is supposed to affect shaders, but the shaderlog is too large for me to find the broken case specifically.
@SakataGintokiYT If the game still runs, could you attach the renderdoc trace for the menu? I already located the missing branch op, but I need to know what is generated in its place with my commits.:

![image](https://cloud.githubusercontent.com/assets/15904127/21749007/1ee9f594-d5a5-11e6-8e53-c72496c1f2f6.png)

**Comment by SakataGintokiYT** [comment]
[21e3f31]
@kd-11 
**[Shaderlog:](https://mega.nz/#!xVZE0JzI!WdeqpQL14AD-BXs6-rEt2xZ1HCoLZKl2M4OI4c2Y6Ko)**
**[RenderDoc-Menu:](https://mega.nz/#!tZJBVTrA!R9qdGXk0sY2TYMlR1dY4wWXTgSzzes9_2ru4kQiKeP0)**

**Comment by kd-11** [comment]
Nice. The test is now in the right place, just has the wrong condition.
Use https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.386 and attach the trace. It should write more info into the shader itself.

**Comment by SakataGintokiYT** [comment]
[236d51ed]

@kd-11 
**[Shaderlog:](https://mega.nz/#!IUJGXKqC!cQfQz2Ro3d3oGXcBfTg7vtImisgHLBhVm7Zi0-Gx7v0)**
**[RenderDoc:](https://mega.nz/#!xUJSDZ4b!KVd9pAkeHRcMena5ui9-vSlO2oQSLcF4zNY4rhKS2hg)**

**Comment by kd-11** [comment]
I think https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.387 should fix this one. From analyzing different instances where the instruction appears, and taking into account known and used bits, it seems bit 25 switches on/off whether the branch will be taken. The shaders given by this game have more than one way to compute position and the branch should determine which path to take.
I have considered the possibility that it may be a multi-bit index into a constants register array, but looking at vertex constants, it seems unlikely as they are used for other computations and contain float data.

**Comment by kd-11** [comment]
Another alternative would be that we should decode input as a component of vertex input.

**Comment by SakataGintokiYT** [comment]
[4a855794]
@kd-11 

**[Saint Seiya Brave Soldiers]**
**[*Video:](https://www.youtube.com/watch?v=NSvHPVv5iZY)**
**[*Shaderlog:](https://mega.nz/#!RYACTIpZ!BCh1r9xC9YUpF8n02aojk3jlo985I3-LOMsxMlQ37a8)**
**[*RenderDoc-In game:](https://mega.nz/#!8QgyVSBb!cNNZQzaXgZ10eJfPDGqwy4VlHQsjEp-OOXKg2QfAxLM)**
**[*RenderDoc-Menu:](https://mega.nz/#!MNJxBbgY!gZFVgHzgUFIS3_m0bUXqxawH4YhMVumoq-VTXwuVUpw)**

---------------------------------------------------------------------------------------------------------------

**[Saint Seiya: Soldiers' Soul]**
**[*Video:](https://www.youtube.com/watch?v=PF3rLlxCeQc)**
**[*Shaderlog:](https://mega.nz/#!dYAyTQqI!D9SrL19M0u3Ortk_7bFRKULp0L9wy8uxw-lp7fn4hTk)**
**[*RenderDoc-Menu:](https://mega.nz/#!ddBnzAZA!ojl1SeZnv25SAHGLOpn8oVJZ0R5400edKBJhgx8oS1g)**
**[*RenderDoc-In game:](https://mega.nz/#!lBI30CBL!hI1rvnSN6DLmImV1H_7DLlji8b0ChOegx7-0Y4bip14)**

**Comment by ikki84** [comment]
@kd-11 If you revert this pull https://github.com/RPCS3/rpcs3/pull/2206 , and keep the posts pull requests after #2206  The game will render properly.
I don´t know exactly which part of the pull make the things flicker and render wrong but my hint is that its related to Persistent mapped buffers.

**Comment by SakataGintokiYT** [comment]
Same without this commit #2206 / I have Nvidia GPU

**Comment by ikki84** [comment]
@SakataGintokiYT I understand it, but I think if @kd-11 examine that pull, he could attempt to fix it for Nvidia users too.

![saintseiya1](https://cloud.githubusercontent.com/assets/16824659/21754264/af1d9504-d5db-11e6-8a9c-3d9c7ffaec22.jpg)

**Comment by kd-11** [comment]
In that case, please test with legacy opengl buffers. If that fixes the flicker, I know why persistent buffers are not operating in an optimal manner. My assumption was that the buffers would be large enough that waits would happen only once every few seconds (there's a flush command in there for god's sake which harms performance). Looking at the renderdoc traces, its obvious that there are very many draw calls per frame and obviously trampling would take place. I shall fix the persistent buffers bug ASAP.

**Comment by kd-11** [comment]
Okay, after looking at the renderdoc trace, it seems I messed up the commit. This issue ought to have been fixed; sorry for the confusion.

**Comment by kd-11** [comment]
https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.389 should fix the opcode issue. As for the buffer RAW hazard, I expect some corruption but only when persistent mapped buffers are enabled and the issue would not be easy to capture using a trace, manifesting as an unstable render.

**Comment by MarioSonic2987** [comment]
Fixed in https://github.com/kd-11/rpcs3/commit/7a758f4a98328e5daefb18e071086bb0ca56aec9!
![](http://i.imgur.com/T8BzpBE.jpg)
![](http://i.imgur.com/4fPH4Vv.jpg)
In Vulkan too.

Also, there's a color error in Vulkan
![](http://i.imgur.com/EP0ZiIV.png)
![](http://i.imgur.com/HpOshIb.png)

**Comment by MarioSonic2987** [comment]
Video: https://youtu.be/JN_RihPI6WI

**Comment by kd-11** [comment]
Seems to be another channel remap issue for the colors. I'll clean up the decompiler code and submit for merge into master. The vulkan colors thing should be easy to fix.
Can someone please attach the D3D12 shader log? I'll try and fix that compilation error if the broken shader is captured in the logs.

**Comment by SakataGintokiYT** [comment]
[7a758f4a]

@kd-11 

```
E {rsx::thread} RSX: FS build failed:0000000007192fc0
F {rsx::thread} class std::runtime_error thrown: fragment program compilation failure
(in file c:\rpcs3\rpcs3\emu\rsx\d3d12\D3D12PipelineState.h:162)
```
**[D3D12-Shaderlog](https://mega.nz/#!dNwHlZLZ!S0yXWMjgtwvkXFzkdTsQGgleODkJtwbl7zflcKhO65I)**

**Comment by kd-11** [comment]
Check if https://ci.appveyor.com/project/kd-11/rpcs3/build/1.0.390 introduces any regressions to vulkan or opengl. I've changed the BRB implementation a bit.

**Comment by MarioSonic2987** [comment]
In https://github.com/kd-11/rpcs3/commit/247fe3d51ac25801d2b61d26ac86684acf1243e2, there's no regression. 
![](http://i.imgur.com/uE5UGL3.jpg)

**Comment by ikki84** [comment]
@kd-11 , Sorry to bother you. After working in https://github.com/RPCS3/rpcs3/pull/2391, is possible to review this case again?

**Comment by kd-11** [comment]
@ikki84 That's the plan. I temporarily abandoned this one since it breaks some games, but I'll focus on it once the current stuff is merged.

**Comment by Zangetsu38** [comment]
@kd-11 i have retested with you last commit, and have a see problem, same splatterhouse, game very pixelisize, but only in render 3d, no for HUD
just see.
![capture d ecran 2017-02-17 15 59 42](https://cloud.githubusercontent.com/assets/5261759/23070304/fad78a36-f52a-11e6-95c9-f9932b0aadd7.png)
![capture d ecran 2017-02-17 16 01 28](https://cloud.githubusercontent.com/assets/5261759/23070305/fbbe2fcc-f52a-11e6-8864-ece81aa9b29b.png)
![capture d ecran 2017-02-17 16 06 15](https://cloud.githubusercontent.com/assets/5261759/23070325/0c30e9bc-f52b-11e6-80cf-fa80250af2c6.png)

**Comment by kd-11** [comment]
Current software upscaling is very rough. I'm avoiding using bilinear filtering due to performance concerns, but will polish up before merging. The image would still not be as sharp as a native image though. Does the log show an "unsupported RTT scaling" warning?

**Comment by Zangetsu38** [comment]
No, i don't see this 
[SSSB.txt](https://github.com/RPCS3/rpcs3/files/783976/SSSB.txt)

**Comment by Zangetsu38** [comment]
@kd-11 Ico HD show this in Loop
>E {rsx::thread} RSX: Unsupported RTT scaling factor: dst_pitch=5120 src_pitch=4864

**Comment by kd-11** [comment]
With latest commit, that error is just a warning and the image is copied without scaling. I'd expect maybe extra padding on the right. Does it output correct images?

**Comment by Zangetsu38** [comment]
Ico show nothing, black screen

**Comment by kd-11** [comment]
Please retest BRB issue with https://github.com/RPCS3/rpcs3/pull/2426

**Comment by ikki84** [comment]
SSBS: Fixed in OpenGL (rendering both character and armor) also rendering in Vulkan.
Crashing in DX12.

**Comment by SakataGintokiYT** [comment]
@kd-11 
```
**DX12**
Unhandled Win32 exception 0xC0000005.
Segfault reading location ffffffffffffffff at 00000000003bd400.
Instruction address: 00000000003bd400.
Module name: 'rpcs3.exe'.
Module base: 0000000000010000.
RPCS3 image base: 0000000000010000.
```
**[Log:](https://gist.github.com/SakataGintokiYT/0bd7aa313a73393909e7d307d016f084)**
**[Shaderlog:](https://mega.nz/#!ANQjXIaL!azTdKDaYh7XiQOFr32hhi4PPSGqXaRzgZjc6BMB_iBk)**

**Comment by raven02** [comment]
Tested with https://github.com/RPCS3/rpcs3/pull/2426 

In summary,

GL : works prefectly fine 
DX12: crashes at start up
Vulkan : Vague chacracter and wrong logo color (YEBIS2)

**Comment by kd-11** [comment]
I've fixed some DX12 crashes, so you can retest with newer commit

**Comment by SakataGintokiYT** [comment]
@kd-11 
```
E {rsx::thread} RSX: FS build failed:D:\EMU\PS3\RPCS3.V2\FragmentProgram.hlsl(197,12-55): warning X3570: gradient instruction used in a loop with varying iteration, attempting to unroll the loop
D:\EMU\PS3\RPCS3.V2\FragmentProgram.hlsl(188,3-36): error X3511: unable to unroll loop, loop does not appear to terminate in a timely manner (248 iterations) or unrolled loop is too large, use the [unroll(n)] attribute to force an exact higher number

F {rsx::thread} class std::runtime_error thrown: fragment program compilation failure
(in file c:\rpcs3\rpcs3\emu\rsx\d3d12\D3D12PipelineState.h:162)

```

**Comment by kd-11** [comment]
That's not related to BRB which is a vertex shader instruction. I think that wraps it up for BRB then.
